src/inc/internalunknownimpl.h

```diff
@@ -69,7 +69,10 @@ namespace ComUtil
         typedef char (&_Yes)[1];
         typedef char (&_No)[2];
 
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-function"
         static inline _No _IsTypeWrapper(...);
+#pragma GCC diagnostic pop
```

Comment:
@jkotas, @janvorli, @franksinankaya 

I could not figure out a better way to fix this one.

Original error states:

```sh
/coreclr/src/inc/internalunknownimpl.h:72:27: error: 'char (& ComUtil::detail::_IsTypeWrapper(...))[2]' declared 'static' but never defined [-Werror=unused-function]
         static inline _No _IsTypeWrapper(...);
                           ^~~~~~~~~~~~~~
```

If we remove `inline`, it is exactly the same error.

If we add `template <typename T>` (as its overload two line below), then g++ (v7.3 on Ubuntu 18.04) fails to deduce template:

```sh
In file included from /coreclr/src/vm/coreclr/../../binder/inc/coreclrbindercommon.h:10:0,
                 from /coreclr/src/vm/coreclr/../../binder/inc/clrprivbindercoreclr.h:9,
                 from /coreclr/src/vm/coreclr/../../binder/inc/assembly.hpp:24,
                 from /coreclr/src/vm/coreclr/corebindresult.h:18,
                 from /coreclr/src/vm/assemblyspecbase.h:19,
                 from /coreclr/src/vm/pefile.h:28,
                 from /coreclr/src/strongname/api/common.h:323,
                 from /coreclr/src/strongname/api/strongname.cpp:11:
/coreclr/src/inc/internalunknownimpl.h: In instantiation of 'const bool ComUtil::IsTypeWrapper<ICLRPrivBinder>::value':
/coreclr/src/inc/internalunknownimpl.h:131:16:   required from 'struct ComUtil::detail::DeriveTypeList<mpl::type_list<ICLRPrivBinder, mpl::null_type> >'
/coreclr/src/inc/internalunknownimpl.h:308:11:   required from 'class ComUtil::IUnknownCommon<ICLRPrivBinder>'
/coreclr/src/vm/coreclr/../../binder/inc/clrprivbindercoreclr.h:17:37:   required from here
/coreclr/src/inc/internalunknownimpl.h:85:42: error: no matching function for call to '_IsTypeWrapper(ICLRPrivBinder*)'
             sizeof(detail::_IsTypeWrapper((T*)0)) == sizeof(detail::_Yes)>::value;
                   ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
/coreclr/src/inc/internalunknownimpl.h:73:27: note: candidate: template<class T> char (& ComUtil::detail::_IsTypeWrapper(...))[2]
         static inline _No _IsTypeWrapper(...);
                           ^~~~~~~~~~~~~~
/coreclr/src/inc/internalunknownimpl.h:73:27: note:   template argument deduction/substitution failed:
/coreclr/src/inc/internalunknownimpl.h:85:42: note:   couldn't deduce template parameter 'T'
             sizeof(detail::_IsTypeWrapper((T*)0)) == sizeof(detail::_Yes)>::value;
                   ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~
/coreclr/src/inc/internalunknownimpl.h:76:21: note: candidate: template<class T> char (& ComUtil::detail::_IsTypeWrapper(T*, typename T::wrapped_type*))[1]
         static _Yes _IsTypeWrapper(T *, typename T::wrapped_type * = nullptr);
                     ^~~~~~~~~~~~~~
/coreclr/src/inc/internalunknownimpl.h:76:21: note:   template argument deduction/substitution failed:
/coreclr/src/inc/internalunknownimpl.h: In substitution of 'template<class T> char (& ComUtil::detail::_IsTypeWrapper(T*, typename T::wrapped_type*))[1] [with T = ICLRPrivBinder]':
/coreclr/src/inc/internalunknownimpl.h:85:42:   required from 'const bool ComUtil::IsTypeWrapper<ICLRPrivBinder>::value'
/coreclr/src/inc/internalunknownimpl.h:131:16:   required from 'struct ComUtil::detail::DeriveTypeList<mpl::type_list<ICLRPrivBinder, mpl::null_type> >'
/coreclr/src/inc/internalunknownimpl.h:308:11:   required from 'class ComUtil::IUnknownCommon<ICLRPrivBinder>'
/coreclr/src/vm/coreclr/../../binder/inc/clrprivbindercoreclr.h:17:37:   required from here
/coreclr/src/inc/internalunknownimpl.h:76:21: error: no type named 'wrapped_type' in 'struct ICLRPrivBinder'
```